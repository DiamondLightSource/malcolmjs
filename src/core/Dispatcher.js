/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
