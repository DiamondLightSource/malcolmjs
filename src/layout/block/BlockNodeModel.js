import { NodeModel } from 'storm-react-diagrams';
import MalcolmPortModel from '../blockPort/blockPort.model';

class BlockNodeModel extends NodeModel {
  constructor(label, description, mri) {
    super('malcolmjsblock');

    this.id = mri;
    this.label = label;
    this.description = description;
    this.addBlockPort = this.addBlockPort.bind(this);
    this.addIcon = this.addIcon.bind(this);
    this.addClickHandler = this.addClickHandler.bind(this);
  }

  addBlockPort(port, portMouseDown) {
    const newPort = new MalcolmPortModel(
      port.input,
      `${this.id}-${port.label}`,
      port.label,
      `${this.id}-${port.label}`
    );

    newPort.addMouseDownHandler(portMouseDown);
    return this.addPort(newPort);
  }

  addIcon(icon) {
    this.icon = icon;
  }

  addClickHandler(handler) {
    this.clickHandler = e => {
      handler(this);
      e.stopPropagation();
    };
  }
}

export default BlockNodeModel;
