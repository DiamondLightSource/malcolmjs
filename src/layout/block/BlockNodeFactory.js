import { AbstractNodeFactory } from 'storm-react-diagrams';
import React from 'react';
import BlockWidget from './blockWidget.component';
import BlockNodeModel from './BlockNodeModel';

/* eslint class-methods-use-this: ["error", { "exceptMethods": ["generateReactWidget", "getNewInstance"] }] */
export default class BlockNodeFactory extends AbstractNodeFactory {
  constructor() {
    super('malcolmjsblock');
  }

  generateReactWidget(diagramEngine, node) {
    return <BlockWidget node={node} />;
  }

  getNewInstance() {
    return new BlockNodeModel();
  }
}
