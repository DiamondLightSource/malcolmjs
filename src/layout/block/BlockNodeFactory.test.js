import { createShallow } from '@material-ui/core/test-utils';
import BlockNodeFactory from './BlockNodeFactory';
import BlockNodeModel from './BlockNodeModel';

describe('BlockWidgetFactory', () => {
  let shallow;
  const blockWidgetFactory = new BlockNodeFactory();

  beforeEach(() => {
    shallow = createShallow();
  });

  it('generates react widget', () => {
    const node = {};
    const wrapper = shallow(blockWidgetFactory.generateReactWidget(null, node));
    expect(wrapper).toMatchSnapshot();
  });

  it('generates new instance', () => {
    const model = blockWidgetFactory.getNewInstance();
    expect(model).toBeInstanceOf(BlockNodeModel);
  });
});
