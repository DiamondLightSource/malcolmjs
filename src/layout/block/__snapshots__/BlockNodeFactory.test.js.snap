// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockWidgetFactory generates react widget 1`] = `
<BlockWidget
  classes={
    Object {
      "block": "BlockWidget-block-1",
      "blockContents": "BlockWidget-blockContents-3",
      "description": "BlockWidget-description-8",
      "iconContents": "BlockWidget-iconContents-7",
      "inputPortsContainer": "BlockWidget-inputPortsContainer-4",
      "outputPortsContainer": "BlockWidget-outputPortsContainer-5",
      "portContainer": "BlockWidget-portContainer-6",
      "title": "BlockWidget-title-2",
    }
  }
  node={Object {}}
/>
`;
